src/metabase/models/collection.clj

```diff
@@ -351,7 +351,7 @@
   highest-level (e.g. most distant) ancestor."
   [{:keys [location]}]
   (when-let [ancestor-ids (seq (location-path->ids location))]
-    (db/select [Collection :name :id] :id [:in ancestor-ids] {:order-by [:%lower.name]})))
+    (db/select [Collection :name :id] :id [:in ancestor-ids] {:order-by [:location]})))
```

Comment:
This looks like this actually does what the docstrings says after your change. It makes me wonder why there was that mismatch between docstring and behavior in the first place. Git blame says I wrote it all of this stuff at the same time about 4 and a half years ago. I don't remember why. Maybe we wanted it sorted by Collection name in the frontend, but I'm not sure where Collection **ancestors** would be used in the FE.